pandas/core/indexes/base.py

```diff
@@ -112,14 +112,18 @@ class Index(IndexOpsMixin, PandasObject):
     dtype : NumPy dtype (default: object)
     copy : bool
         Make a copy of input ndarray
-    name : object
+    name : object, optional
         Name to be stored in the index
+    names : sequence of objects, optional
+        Names for the index levels used when attempt to create a MultiIndex
```

Comment:
I don't think this is correct - in this version of the PR, it can be used even when a ``MultiIndex`` is not created.

Reply:
@toobaz  I can see what you mean, thanks. Personally, I think it's worse clarifying that the purpose of `names` is for creation a `MultiIndex` at the end of the day.
What do you think is the best way to describe this parameter then?